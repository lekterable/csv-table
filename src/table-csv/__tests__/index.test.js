import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'
import Table from '../'

describe('table-csv', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const component = renderer.create(<Table />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('should match snapshot with data', () => {
    const component = renderer
      .create(
        <Table data={[['Header 1', 'Header 2'], ['Value 1', 'Value 2']]} />
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('should not render <Row /> without data', () => {
    const { queryAllByTestId } = render(<Table />)
    const header = queryAllByTestId('table-row')
    const rows = queryAllByTestId('table-row')

    expect(header.length).toBe(0)
    expect(rows.length).toBe(0)
  })

  it('should render <Row /> with data', () => {
    const { queryAllByTestId } = render(
      <Table data={[['Header 1', 'Header 2'], ['Value 1', 'Value 2']]} />
    )
    const header = queryAllByTestId('table-header-row')
    const rows = queryAllByTestId('table-row')

    expect(header.length).toBe(1)
    expect(rows.length).toBe(1)
  })

  it('should not render <Toolbar />', () => {
    const { queryByTestId } = render(<Table />)
    const toolbar = queryByTestId('toolbar')

    expect(toolbar).toBe(null)
  })

  it('should render <Toolbar /> if `toolbar` prop', () => {
    const { queryByTestId } = render(<Table toolbar />)
    const toolbar = queryByTestId('toolbar')

    expect(toolbar).not.toBe(null)
  })

  it('should not render <ExportButton />', () => {
    const { queryByTestId } = render(<Table />)
    const exportButton = queryByTestId('export-button')

    expect(exportButton).toBe(null)
  })

  it('should render <ExportButton /> if `exportable` prop', () => {
    const { queryByTestId } = render(<Table exportable />)
    const exportButton = queryByTestId('export-button')

    expect(exportButton).not.toBe(null)
  })
})
