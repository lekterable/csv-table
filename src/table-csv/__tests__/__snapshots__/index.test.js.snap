// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table-csv should match snapshot 1`] = `
<div
  className="csv-table-wrapper"
>
  <div
    className="features"
  />
  <table
    className="csv-table"
  >
    <thead
      className="csv-table__header"
    />
    <tbody
      className="csv-table__body"
    />
  </table>
</div>
`;

exports[`table-csv should match snapshot with data 1`] = `
<div
  className="csv-table-wrapper"
>
  <div
    className="features"
  />
  <table
    className="csv-table"
  >
    <thead
      className="csv-table__header"
    >
      <tr
        className="table__row"
        data-testid="table-header-row"
      >
        <th
          className="table__cell table__cell--head"
          data-testid="table-header-cell"
        >
          <div
            className="editable-field"
            data-testid="editable-field-display"
            onClick={[Function]}
          >
            Header 1
          </div>
        </th>
        <th
          className="table__cell table__cell--head"
          data-testid="table-header-cell"
        >
          <div
            className="editable-field"
            data-testid="editable-field-display"
            onClick={[Function]}
          >
            Header 2
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      className="csv-table__body"
    >
      <tr
        className="table__row"
        data-testid="table-row"
      >
        <td
          className="table__cell"
          data-testid="table-cell"
        >
          <div
            className="editable-field"
            data-testid="editable-field-display"
            onClick={[Function]}
          >
            Value 1
          </div>
        </td>
        <td
          className="table__cell"
          data-testid="table-cell"
        >
          <div
            className="editable-field"
            data-testid="editable-field-display"
            onClick={[Function]}
          >
            Value 2
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
